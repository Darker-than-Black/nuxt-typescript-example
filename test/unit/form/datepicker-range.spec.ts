import DatePicker from 'vue2-datepicker'
import VueTestInstance from '~/test/factory/vue-instance'
import DatepickerRange from '~/components/form/datepicker-range.vue'

describe('datepicker-range.vue', () => {
  let wrapper: any
  beforeEach(() => {
    wrapper = VueTestInstance(DatepickerRange, {
      isShallow: false,
      modules: [DatePicker],
      options: {
        props: {
          value: []
        }
      }
    })
  })

  it('initialize correctly', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('check input function', () => {
    const date: string[] = ['19.07.20', '20.08.20']
    expect(wrapper.vm.input(date)).toEqual(date)
  })
})
