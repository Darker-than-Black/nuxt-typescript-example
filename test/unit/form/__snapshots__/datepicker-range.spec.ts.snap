// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`datepicker-range.vue should match the snapshot 1`] = `
<div><label for="" class="title__field">

    <!----></label>
  <div>
    <div class="datepicker mx-datepicker mx-datepicker-range">
      <div class="mx-input-wrapper"><input placeholder="" name="date" type="text" autocomplete="off" class="input input__date">
        <!----> <i class="mx-icon-calendar"><img src="/icons/Calendar.svg" alt="Calendar" class="img"></i></div>
    </div>
  </div>
</div>
`;
